account/account_test.go

```diff
@@ -344,3 +355,132 @@ func TestConvertGDPREnforcePurposeFields(t *testing.T) {
 		}
 	}
 }
+
+func TestGdprCcpaChannelEnabled(t *testing.T) {
+	cfg := &config.Configuration{}
+	fetcher := &mockAccountFetcher{}
+	assert.NoError(t, cfg.MarshalAccountDefaults())
+
+	testCases := []struct {
+		description         string
+		givenAccountID      string
+		givenMetric         string
+		expectedMetricCount int
+	}{
+		{
+			description:         "Account is GDPR Channel Enabled, the GDPR metric should be called",
+			givenAccountID:      "gdpr_channel_enabled_acct",
+			givenMetric:         "RecordAccountGDPRChannelEnabledWarning",
+			expectedMetricCount: 1,
+		},
+		{
+			description:         "Account is CCPA Channel Enabled, the CCPA metric should be called",
+			givenAccountID:      "ccpa_channel_enabled_acct",
+			givenMetric:         "RecordAccountCCPAChannelEnabledWarning",
+			expectedMetricCount: 1,
+		},
+		{
+			description:         "Account isn't channel enabled, CCPA metric shouldn't be called",
+			givenAccountID:      "valid_acct",
+			givenMetric:         "RecordAccountCCPAChannelEnabledWarning",
+			expectedMetricCount: 0,
+		},
+		{
+			description:         "Account isn't channel enabled, GDPR metric shouldn't be called",
+			givenAccountID:      "valid_acct",
+			givenMetric:         "RecordAccountGDPRChannelEnabledWarning",
+			expectedMetricCount: 0,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.description, func(t *testing.T) {
+			metrics := &metrics.MetricsEngineMock{}
+			metrics.Mock.On(test.givenMetric, mock.Anything, mock.Anything).Return()
+			metrics.Mock.On("RecordAccountUpgradeStatus", mock.Anything, mock.Anything).Return()
+
+			_, _ = GetAccount(context.Background(), cfg, fetcher, test.givenAccountID, metrics)
+
+			metrics.AssertNumberOfCalls(t, test.givenMetric, test.expectedMetricCount)
+			metrics.AssertNumberOfCalls(t, "RecordAccountUpgradeStatus", test.expectedMetricCount)
```

Comment:
Are there no cases where `RecordAccountUpgradeStatus` is called or not called independently of the other given metrics? If not, would it be better to just drop this metrics and replace it with a sum over the other metrics?